packages/cli/src/webfinger/error.ts

```diff
@@ -0,0 +1,47 @@
+import { type Message, message } from "@optique/core";
+
+/**
+ * Generates a user-friendly error message based on the type of error
+ * encountered during WebFinger lookup.
+ * @param {string} resource The resource being looked up.
+ * @param {unknown} error The error encountered.
+ * @returns {string} A descriptive error message.
+ */
+export const getErrorMessage = (resource: string, error: unknown): Message =>
+  error instanceof InvalidHandleError
+    ? message`Invalid handle format: ${error.handle}`
+    : error instanceof NotFoundError
+    ? message`Resource not found: ${error.resource}`
+    : error instanceof Error
+    ? message`Error looking up WebFinger for ${resource}: ${error.message}`
+    : message`Error looking up WebFinger for ${resource}: ${String(error)}`;
```

Comment:
It's trivial, but as `printError()` prepends `Error: ` by default, it might sound unnatural: `Error: Error looking up WebFinger for ...: ...`.

```suggestion
    ? message`Failed to look up WebFinger for ${resource}: ${error.message}`
    : message`Failed to look up WebFinger for ${resource}: ${String(error)}`;
```